### PR TITLE
🎨 Improve format filtering.

### PR DESCRIPTION
**Pull request by matthiasrohmer**

### CODE REVIEW

**Comment by matthiasrohmer** [comment]
Will extend this PR by something that removes empty level-1 sidebar items.

**Comment by matthiasrohmer** [comment]
Leading tutorial dividers are now also removed, as well as empty first level categories. Additionally I fixed the issues with the dropdown you mentioned, should work a lot better now.

File: platform/lib/common/filteredPage.js [hunk 1]
```diff
@@ -122,9 +121,22 @@ class FilteredPage {
...
+          if (!navItem.has('a').length) {
+            navItem.remove();
+          }
+        });
+
     // Remove eventually unnecessary tutorial dividers left by the
     // previous transformation
-    this._dom('.nav-item-tutorial-divider:last-child').remove();
+    /* eslint-disable max-len */
+    this._dom('.nav-item-tutorial-divider:last-child, .nav-item-tutorial-divider:first-child').remove();
```
Comment (sebastianbenz) [inline platform/lib/common/filteredPage.js]: nit: I'd prefer splitting the string over disabling eslint.

File: platform/lib/common/filteredPage.js [hunk 2]
```diff
@@ -148,13 +160,15 @@ class FilteredPage {
     // Rewrite the active state (which is websites per default) to
     // the current active format
     const activeFormat = this._dom('.ap-m-format-toggle-selected');
-    if (!activeFormat.html()) {
-      console.error('doc did not specify an active format');
+    if (activeFormat.length == 0) {
+      throw new Error('Page has no active format.');
```
Comment (sebastianbenz) [inline platform/lib/common/filteredPage.js]: I don't like throwing an exception here, I it happens in prod the user gets a 502, but it's something we can recover easily from.

Comment (matthiasrohmer) [reply platform/lib/common/filteredPage.js]: Ah you are right. I only had the implementation for local in mind, the error is catched there. Will revert.

File: platform/lib/common/filteredPage.js (hunk 1 above)
Comment (matthiasrohmer) [reply platform/lib/common/filteredPage.js]: Definitely the better solution 😅

**Pull request merged**